### PR TITLE
Add neo4j-driver 4.4.5

### PR DESCRIPTION
**Pull request by rxm7706**
Dependency for amundsen

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by rxm7706** [comment]
@conda-forge/staged-recipes  ready for review

**Comment by rxm7706** [comment]
https://github.com/neo4j/neo4j-python-driver/blob/5.0/LICENSE.PYTHON.txt

`Python software and documentation are licensed under the
Python Software Foundation License Version 2.

Starting with Python 3.8.6, examples, recipes, and other code in
the documentation are dual licensed under the PSF License Version 2
and the Zero-Clause BSD license.`

But linter complains about https://spdx.org/licenses/PSF-2.0.html

File: recipes/neo4j-driver/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    - testkitbackend
+  commands:
+    - pip check
+  requires:
+    - pip
+
+about:
+  home: https://github.com/neo4j/neo4j-python-driver
+  summary: Neo4j Bolt driver for Python
+  license: Apache-2.0 & PSF-2.0
```
Comment (dbast) [inline recipes/neo4j-driver/meta.yaml:38]:
```suggestion
  license: Apache-2.0 AND PSF-2.0
```

File: recipes/neo4j-driver/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    - neo4j-python-driver
+    - pip
+  run:
+    - python >=3.6
+    - pytz
+
+test:
+  imports:
+    - neo4j
+    - testkitbackend
```
Comment (xhochy) [inline recipes/neo4j-driver/meta.yaml:29]: This sounds like testing code is erroneously packaged. Can you delete this and report it upstream?

Comment (rxm7706) [reply recipes/neo4j-driver/meta.yaml]: - testkitbackend is not added when I use grayskull - removed from meta.yaml

**Comment by dbast** [comment]
from https://github.com/neo4j/neo4j-python-driver#installation

```
neo4j-driver is the old name for this package. It is now deprecated and and will receive no further updates starting with 6.0.0. Make sure to install neo4j as shown above.
```

**Comment by rxm7706** [comment]
> from https://github.com/neo4j/neo4j-python-driver#installation
> 
> ```
> neo4j-driver is the old name for this package. It is now deprecated and and will receive no further updates starting with 6.0.0. Make sure to install neo4j as shown above.
> ```

@dbast thank you I am trying to get to Amundsen Metadata and Amundsen Databuilder 
grayskull gives me     - neo4j-python-driver ==1.7.6 while https://github.com/amundsen-io/amundsen/blob/main/metadata/requirements.txt gives me neo4j==4.4.5, however https://github.com/amundsen-io/amundsen/blob/main/databuilder/requirements.txt uses neo4j-driver>=4.4.5,<5.0 - let me see how i can navigate my way through this.

File: recipes/neo4j-driver/meta.yaml (hunk 2 above)
Comment (ocefpaf) [reply recipes/neo4j-driver/meta.yaml]: They extra module is still there. BTW, that was fixed in the latest version. If you need this older version you will need to manually delete it like @xhochy requested above.

**Pull request closed without merging**